v2/blockstore/example_test.go

```diff
@@ -0,0 +1,146 @@
+	// 	bafy2bzaced4ueelaegfs5fqu4tzsh6ywbbpfk3cxppupmxfdhbpbhzawfw5oy -> 821 bytes
+	// 	bafy2bzaceaycv7jhaegckatnncu5yugzkrnzeqsppzegufr35lroxxnsnpspu -> 1053 bytes
+	// 	bafy2bzaceb62wdepofqu34afqhbcn4a7jziwblt2ih5hhqqm6zitd3qpzhdp4 -> 1094 bytes
+	// 	bafy2bzaceb3utcspm5jqcdqpih3ztbaztv7yunzkiyfq7up7xmokpxemwgu5u -> 1051 bytes
+	// 	bafy2bzacedjwekyjresrwjqj4n2r5bnuuu3klncgjo2r3slsp6wgqb37sz4ck -> 821 bytes
+}
+
+// ExampleOpenReadWrite creates a read-write blockstore and puts
+func ExampleOpenReadWrite() {
+	thisBlock := merkledag.NewRawNode([]byte("fish")).Block
+	thatBlock := merkledag.NewRawNode([]byte("lobster")).Block
+	andTheOtherBlock := merkledag.NewRawNode([]byte("barreleye")).Block
+
+	dest := "../testdata/sample-rw-bs-v2.car"
+	roots := []cid.Cid{thisBlock.Cid(), thatBlock.Cid(), andTheOtherBlock.Cid()}
+
+	rwbs, err := blockstore.OpenReadWrite(dest, roots, carv2.UseDataPadding(1413), carv2.UseIndexPadding(42))
+	if err != nil {
+		panic(err)
+	}
+	defer rwbs.Close()
+
+	// Put all blocks onto the blockstore.
+	blocks := []blocks.Block{thisBlock, thatBlock}
+	if err := rwbs.PutMany(blocks); err != nil {
+		panic(err)
+	}
+	fmt.Printf("Successfully wrote %v blocks into the blockstore.\n", len(blocks))
+
+	// Any blocks put can be read back using the same blockstore instance.
+	block, err := rwbs.Get(thatBlock.Cid())
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("Read back block just put with raw value of `%v`.\n", string(block.RawData()))
+
+	// Finalize the blockstore to flush out the index and make a complete CARv2.
+	if err = rwbs.Finalize(); err != nil {
+		panic(err)
+	}
+
+	// Resume from the same file to add more blocks.
+	// Note the UseDataPadding and roots must match the values passed to the blockstore instance
+	// that created the original file. Otherwise, we cannot resume from the same file.
+	resumedRwbos, err := blockstore.OpenReadWrite(dest, roots, carv2.UseDataPadding(1413))
+	if err != nil {
+		panic(err)
+	}
+	defer resumedRwbos.Close()
+
+	// Put another block, appending it to the set of blocks that are written previously.
+	if err = resumedRwbos.Put(andTheOtherBlock); err != nil {
+		panic(err)
+	}
+
+	// Read back the the block put before resumption.
+	// Blocks previously put are present.
+	block, err = resumedRwbos.Get(thatBlock.Cid())
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("Resumed blockstore contains blocks put previously with raw value of `%v`.\n", string(block.RawData()))
+
+	// Put an additional block to the CAR.
+	// Blocks put after resumption are also present.
+	block, err = resumedRwbos.Get(andTheOtherBlock.Cid())
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("It also contains the block put after resumption with raw value of `%v`.\n", string(block.RawData()))
+
+	// Finalize the blockstore to flush out the index and make a complete CARv2.
+	// Note, Finalize must be called on an open ReadWrite blockstore to flush out a complete CARv2.
+	if err = resumedRwbos.Finalize(); err != nil {
```

Comment:
nit: this can be `:=`; it helps keep the code simpler to follow as you're not modifying the var in the parent scope. this can be done for any other place where you only assign an error.